Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/resthandler.go

```diff
@@ -328,6 +328,11 @@ func finishRequest(timeout time.Duration, fn resultFunc) (result runtime.Object,
 // setSelfLink sets the self link of an object (or the child items in a list) to the base URL of the request
 // plus the path and query generated by the provided linkFunc
 func setSelfLink(obj runtime.Object, req *restful.Request, namer ScopeNamer) error {
+	// if the object doesn't have a selfLink, no-op
+	if !runtime.HasSelfLink(obj) {
```

Comment:
----- Original Message -----
> > @@ -328,6 +328,11 @@ func finishRequest(timeout time.Duration, fn
> > resultFunc) (result runtime.Object,
> >  // setSelfLink sets the self link of an object (or the child items in a
> >  list) to the base URL of the request
> >  // plus the path and query generated by the provided linkFunc
> >  func setSelfLink(obj runtime.Object, req *restful.Request, namer
> >  ScopeNamer) error {
> > +	// if the object doesn't have a selfLink, no-op
> > +	if !runtime.HasSelfLink(obj) {
> 
> Ok, so before setting the SelfLink, the name is built.  Building that name
> requires retrieving name and namespace.  ResourceAccessReview has neither of
> those fields.  Right now, that get attempt  calls `Accessor()` which
> produces an error.  That causes a failure before any attempt to SetSelfLink.
> 
> The godoc for `Accessor()`
> (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/meta/meta.go#L33)
> suggests that the failures I see about missing fields are intended and
> intentional.
> 
> That means that unless we change `Accessor`'s contract to not fail like that,
> we aren't going to be guaranteed to ever see the error we want because we'll
> fail when doing thinks like getting names and namespaces.

Accessor should return a typed known error that clients can choose to ignore.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1049/files#r24848688